libsolidity/analysis/ViewPureChecker.cpp

```diff
@@ -251,13 +242,8 @@ void ViewPureChecker::reportMutability(StateMutability _mutability, SourceLocati
 			m_currentFunction->stateMutability() == StateMutability::Pure,
 			""
 		);
-		if (!m_enforceViewWithError && m_currentFunction->stateMutability() == StateMutability::View)
-			m_errorReporter.warning(_location, text);
-		else
-		{
-			m_errors = true;
-			m_errorReporter.typeError(_location, text);
-		}
+		m_errors = true;
+		m_errorReporter.typeError(_location, text);
```

Comment:
Perhaps it would be better to move this up now (and delete the `text` variable)?